khal/utils.py

```diff
@@ -183,3 +186,74 @@ def relative_timedelta_str(day: dt.date) -> str:
 
 def get_wrapped_text(widget):
     return widget.original_widget.get_edit_text()
+
+
+def human_formatter(format_string, width=None, colors=True):
+    """Create a formatter that formats events to be human readable."""
+    def fmt(rows):
+        single = type(rows) == dict
+        if single:
+            rows = [rows]
+        results = []
+        for row in rows:
+            if 'calendar-color' in row:
+                row['calendar-color'] = get_color(row['calendar-color'])
+
+            s = format_string.format(**row)
+
+            if colors:
+                s += style('', reset=True)
+
+            if width:
+                results += color_wrap(s, width)
+            else:
+                results.append(s)
+        if single:
+            return results[0]
+        else:
+            return results
+    return fmt
+
+
+CONTENT_ATTRIBUTES = ['start', 'start-long', 'start-date', 'start-date-long',
+                      'start-time', 'end', 'end-long', 'end-date', 'end-date-long', 'end-time',
+                      'duration', 'start-full', 'start-long-full', 'start-date-full',
+                      'start-date-long-full', 'start-time-full', 'end-full', 'end-long-full',
+                      'end-date-full', 'end-date-long-full', 'end-time-full', 'duration-full',
+                      'start-style', 'end-style', 'to-style', 'start-end-time-style',
+                      'end-necessary', 'end-necessary-long', 'repeat-symbol', 'repeat-pattern',
+                      'title', 'organizer', 'description', 'location', 'all-day', 'categories',
+                      'uid', 'url', 'calendar', 'calendar-color', 'status', 'cancelled']
+
+
+def json_formatter(fields):
+    """Create a formatter that formats events in JSON."""
+
+    if len(fields) == 1 and fields[0] == 'all':
+        fields = CONTENT_ATTRIBUTES
+
+    def fmt(rows):
+        single = type(rows) == dict
+        if single:
+            rows = [rows]
+
+        filtered = []
+        for row in rows:
+            f = dict(filter(lambda e: e[0] in fields and e[0] in CONTENT_ATTRIBUTES, row.items()))
+
+            if f.get('repeat-symbol', '') != '':
+                f["repeat-symbol"] = f["repeat-symbol"].strip()
+            if f.get('status', '') != '':
+                f["status"] = f["status"].strip()
+            if f.get('cancelled', '') != '':
+                f["cancelled"] = f["cancelled"].strip()
+
+            filtered.append(f)
+
+        results = [json.dumps(filtered, ensure_ascii=False)]
```

Comment:
**non-blocking suggestion**:

Maybe format this with whitespace to make it a bit more human-readable too?

```suggestion
        results = [json.dumps(filtered, indent=2, ensure_ascii=False)]
```